geps/gep-1619/index.md

```diff
@@ -317,71 +346,55 @@ In this section, we will explore the questions and design elements associated wi
 ### GO
 
 ```go
-// SessionPersistencePolicy provides a way to define session persistence rules
-// for a service or route.
+// LoadBalancerPolicy provides a way to define load balancing rules
+// for a service.
 //
-// Support: Core
-type SessionPersistencePolicy struct {
+// Support: Extended
+type LoadBalancerPolicy struct {
     metav1.TypeMeta   `json:",inline"`
     metav1.ObjectMeta `json:"metadata,omitempty"`
 
-    // Spec defines the desired state of SessionPersistencePolicy.
-    Spec SessionPersistencePolicySpec `json:"spec"`
+    // Spec defines the desired state of LoadBalancerPolicy.
+    Spec LoadBalancerPolicySpec `json:"spec"`
 
-    // Status defines the current state of SessionPersistencePolicy.
-    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+    // Status defines the current state of LoadBalancerPolicy.
+    Status LoadBalancerPolicyStatus `json:"status,omitempty"`
 }
 
-// SessionPersistencePolicySpec defines the desired state of
-// SessionPersistencePolicy.
+// LoadBalancerPolicySpec defines the desired state of
+// LoadBalancerPolicy.
 // Note: there is no Override or Default policy configuration.
-type SessionPersistencePolicySpec struct {
+type LoadBalancerPolicySpec struct {
     // TargetRef identifies an API object to apply policy to.
-    // The TargetRef may be a Service, HTTPRoute, GRPCRoute,
-    // or a HTTPRouteRule or GRPCRouteRule section.
-    // At least one of these targets must be supported for
-    // core-level compliance.
-    //
+    // Currently, Services are the only valid API target references.
     TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
 
-    // AbsoluteTimeoutSeconds defines the absolute timeout of the
-    // persistent session measured in seconds. Once
-    // AbsoluteTimeoutSeconds has elapsed, the session becomes invalid.
-    //
-    // Support: Core
-    //
-    // +optional
-    AbsoluteTimeoutSeconds int64 `json:"absoluteTimeoutSeconds,omitempty"`
-
-    // IdleTimeoutSeconds defines the idle timeout of the
-    // persistent session measured in seconds. Once the session
-    // has been idle for more than specified IdleTimeoutSeconds
-    // duration, the session becomes invalid.
-    //
-    // Support: Core
-    //
-    // +optional
-    IdleTimeoutSeconds int64 `json:"idleTimeoutSeconds,omitempty"`
+    // SessionPersistence defines and configures session persistence.
+    SessionPersistence *SessionPersistence `json:"sessionPersistence"`
+}
 
+// SessionPersistence defines the desired state of
+// SessionPersistence.
+type SessionPersistence struct {
     // SessionName defines the name of the persistent session token
     // (e.g. a cookie name).
```

Comment:
Can we add 'or a header name' - to make it more clear that it's not restricted to cookies ?